From 112c17e00246d562fc923ec395f124a7101fda3f Mon Sep 17 00:00:00 2001
From: Yegor Yefremov <yegorslists@googlemail.com>
Date: Thu, 10 Mar 2016 09:30:46 +0100
Subject: [PATCH] Fix musl build

Replace __BEGIN_DECLS/__END_DECLS macros with following blocks:

extern "C" {

}

__BEGIN_DECLS/__END_DECLS are not part of all C libraries.

Signed-off-by: Yegor Yefremov <yegorslists@googlemail.com>
---
 src/include/netinet/sctp.h | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/include/netinet/sctp.h b/src/include/netinet/sctp.h
index 51f5bfb..47571cd 100644
--- a/src/include/netinet/sctp.h
+++ b/src/include/netinet/sctp.h
@@ -37,7 +37,9 @@
 #include <linux/types.h>
 #include <sys/socket.h>
 
-__BEGIN_DECLS
+#ifdef __cplusplus
+extern "C" {
+#endif
 
 typedef __s32 sctp_assoc_t;
 
@@ -862,6 +864,8 @@ int sctp_recvmsg(int s, void *msg, size_t len, struct sockaddr *from,
 /* Return the address length for an address family. */
 int sctp_getaddrlen(sa_family_t family);
 
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
 
 #endif /* __linux_sctp_h__ */
-- 
2.7.0

